### PR TITLE
 Examples code updated with changes as requested (default values removed, use of variables restricted to only least, local values used wherever applicable, provider with lowest and one latest version defined, test/plan/apply verified)

### PR DESCRIPTION
**Pull request by marifse**
### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: examples/advanced/pre_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	terraform.InitAndApply(t, terraformOptions)
+
+	// Validate the Cloud Object Storage instance creation.
+	bucket1ModuleID := terraform.Output(t, terraformOptions, "bucket_name1")
+	assert.NotEmpty(t, bucket1ModuleID, "COS instance ID should not be empty")
+
+	// Validate the Cloud Object Storage instance creation.
+	bucket2ModuleID := terraform.Output(t, terraformOptions, "bucket_name2")
+	assert.NotEmpty(t, bucket2ModuleID, "COS instance ID should not be empty")
+}
```
Comment (ocofaigh) [inline examples/advanced/pre_test.go:28]: As mentioned in previous PR, all tests need to be in pr_test.go, or other_test.go which are both in the tests folder. We already do `InitAndApply` tests in there, but perhaps extra assertions could be added like you have here

File: examples/advanced/variables.tf [hunk 2]
```diff
@@ -48,7 +45,6 @@ variable "cross_region_location" {
 variable "management_endpoint_type_for_bucket" {
   type        = string
   description = "The type of endpoint for the IBM terraform provider to use to manage the bucket. (public, private, direct)"
-  default     = "public"
```
Comment (ocofaigh) [inline examples/advanced/variables.tf:51]: please add this back

File: examples/advanced/variables.tf [hunk 3]
```diff
@@ -60,5 +56,4 @@ variable "resource_group" {
 variable "existing_at_instance_crn" {
   type        = string
   description = "Optionally pass an existing activity tracker instance CRN to use in the example. If not passed, a new instance will be provisioned"
-  default     = null
```
Comment (ocofaigh) [inline examples/advanced/variables.tf:63]: please add this back. if `null` is passed, the example will create a new AT instance

File: examples/fscloud/variables.tf [hunk 4]
```diff
@@ -42,7 +39,7 @@ variable "resource_group" {
 variable "existing_at_instance_crn" {
   type        = string
   description = "Optionally pass an existing activity tracker instance CRN to use in the example. If not passed, a new instance will be provisioned"
-  default     = null
```
Comment (ocofaigh) [inline examples/fscloud/variables.tf:45]: please add this back. if `null` is passed, the example will create a new AT instance

File: examples/one-rate-plan/version.tf [hunk 5]
```diff
@@ -6,7 +6,7 @@ terraform {
   required_providers {
     ibm = {
       source  = "ibm-cloud/ibm"
-      version = "1.56.1"
+      version = "1.60.0"
```
Comment (ocofaigh) [inline examples/one-rate-plan/version.tf:9]: please change this back. As per code comment above, this example should lock into the lowest version that provider supports, which in this case is 1.56.1

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Left some comments. Have you set up your local dev environment yet using steps at https://terraform-ibm-modules.github.io/documentation/#/local-dev-setup?id=local-development-setup ? Please do this and run `pre-commit run --all-files` as I think some hooks are going to update some of the formatting in your PR. If so, ensure to commit the changes.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Please also be aware that in every example, you will find a `catalogValidationValues.json.template`. This is used by an internal pipeline for validating the example, so since you are now updating examples variables.tf files with to make some variables required, please ensure this file is also updated.

**Comment by marifse** [comment]
> Left some comments. Have you set up your local dev environment yet using steps at https://terraform-ibm-modules.github.io/documentation/#/local-dev-setup?id=local-development-setup ? Please do this and run `pre-commit run --all-files` as I think some hooks are going to update some of the formatting in your PR. If so, ensure to commit the changes.

@ocofaigh yes i have setup the local enviroment and tested running pre-commit run --all-files, and results are all passed.

**Comment by ocofaigh** [comment]
@marifse thanks, I still dont see any updated to `catalogValidationValues.json.template` ?

**Comment by marifse** [comment]
> @marifse thanks, I still dont see any updated to `catalogValidationValues.json.template` ?

I did not add any new variable in this module examples, though even if i remove the default values do i need to put those variables into catalogValidationValues.json.template file with their hard coded values? I will appreciate your help, as this is the first module on which I am working.

Another thing, on project description i got the instructions to use the one least available provider and one latest available provider in range defined in module as 1.56 and 2.0, while the 1.56.1 is being used in two or three examples, though there is not any example with only latest one 2.0 but two fscloud and advanced examples are with the range defined as >= 1.56.1, < 2.0.0?

**Comment by ocofaigh** [comment]
@marifse If you remove a default value from a variable, it becomes a required variable, so you would then need to update the `catalogValidationValues.json.template` files with a value for the variable, however I just checked and it seems only `prefix` and `region` are being made required, and we already pass values in for them in the json template files, so actually there is no action required here. 

The requirement for testing provider versions is that we need to ensure at least 1 example is testing the lowest version defined the the range, and at least 1 testing whatever the latest version is. So in this case, the supported provider range is `1.56.1, <2.00` which means we would want at least 1 example locked into `1.56.1`, and at least 1 example that will always use the current latest version. To achieve that, you would use `>=1.56.1`.

Generally you will need to lock the most basic example into the lowest supported provider version because some of the more complex example may have dependant module that actually require later provider versions.

**Comment by ocofaigh** [comment]
@marifse The pre-commit hooks are failing in the pipeline? Are you sure they all pass locally for you? Ensure you set up your local dev environment using steps at https://terraform-ibm-modules.github.io/documentation/#/local-dev-setup?id=local-development-setup and then run `pre-commit run --all-files`.

**Pull request closed without merging**